solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -58,29 +50,22 @@ public class FlowReceiverContainer {
 	private final UnsignedCounterBarrier unacknowledgedMessageTracker = new UnsignedCounterBarrier();
 
 	/**
-	 * {@link #rebind(UUID)} can be invoked while this {@link FlowReceiverContainer} is "active"
-	 * (i.e. after a {@link #bind()} and before a {@link #unbind()}).
 	 * Operations which normally expect an active flow to function should use this lock's read lock to seamlessly
 	 * operate as if the flow <b>was not</b> rebinding.
 	 */
 	private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
 	private final Condition bindCondition = readWriteLock.writeLock().newCondition();
```

Comment:
IIRC these 2 fields and the whole `waitForBind()` function were only added so that when the consumer binding calls `receive()` during a rebind (which is a long operation), it wouldn't fail and throw an exception.

But since we're removing rebind acks, these 2 fields and the `waitForBind()` function can now be removed.

Reply:
* Removed waitForBind(). 
* readWriteLock is still needed to serialize calls to bind(). unbind(), pause() and resume(). However no readlock needed and it is removed. 
* removed bindCondition